src/ts/spatialnavigation/navigationgroup.ts

```diff
@@ -0,0 +1,141 @@
+import { Action, Direction } from './spatialnavigation';
+import { Container } from '../components/container';
+import { Component } from '../components/component';
+import { getElementInDirection } from './navigationalgorithm';
+import { getHtmlElementsFromComponents } from './gethtmlelementsfromcomponents';
+import { NodeEventSubscriber } from './nodeeventsubscriber';
+import { isSettingsPanel } from './typeguards';
+
+export type NavigationCallback = (direction: Direction, target: HTMLElement) => boolean;
+export type ActionCallback = (action: Action, target: HTMLElement) => boolean;
+
+export class NavigationGroup {
+  private activeElement?: HTMLElement;
+  private activeElementBeforeDisable?: HTMLElement;
+  private readonly components: Component<unknown>[];
+  private removeElementHoverEventListeners = () => {};
+  private readonly eventSubscriber: NodeEventSubscriber;
+
+  constructor(public readonly container: Container<unknown>, ...components: Component<unknown>[]) {
+    this.components = components;
+    this.eventSubscriber = new NodeEventSubscriber();
+  }
+
+  /**
+   * Adds onNavigation override
+   * @param direction {Direction}
+   * @param target {HTMLElement}
+   *
+   * @returns Boolean - true means it should stop propagation, and false means it should be handled by default handler
+   */
+  public onNavigation?: NavigationCallback;
+
+  /**
+   * Adds onAction override
+   * @param action {Action}
+   * @param target {HTMLElement}
+   *
+   * @returns Boolean - true means it should stop propagation, and false means it should be handled by default handler
+   */
+  public onAction?: ActionCallback;
+
+  public getActiveElement(): HTMLElement | undefined {
+    return this.activeElement;
+  }
+
+  public focusElement(element: HTMLElement): void {
+    this.blurActiveElement();
+    this.activeElement = element;
+    this.activeElement.focus();
+  }
+
+  public blurActiveElement(): void {
+    this.activeElement?.blur();
+    this.activeElement = undefined;
```

Comment:
why is the `activeElement` unset here? Seems a bit confusing.
Maybe just blur it here and unset it in `disable()` instead?

Reply:
Makes sense, addressed in [f97b46b](https://github.com/bitmovin/bitmovin-player-ui/pull/468/commits/f97b46b10c03d2fcae542155352f265bb13a6276)